plugins/modules/ipagroup.py

```diff
@@ -334,8 +381,12 @@ def main():
                         # If yes: modify
                         if not compare_args_ipa(ansible_module, args,
                                                 res_find):
+                            if "nonposix" in args:
```

Comment:
There's no need to remove external. Note that before adding the `group-mod` command the status of external and if it can be change is evaluated, raising an error if the conditions that enable it to change are not met.

Note also, that the `posix` and `nonposix` arguments are removed when they cannot be used, `posix` before `group_add` and `nonposix` before `group_mod`.